src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -269,10 +269,19 @@ def _get_resource_type(self):
             resource_type = command_type.settings.get('resource_type', None) if command_type else None
         return resource_type
 
-    def get_api_version(self, resource_type=None):
+    def get_api_version(self, resource_type=None, operation_group=None):
         from azure.cli.core.profiles import get_api_version
         resource_type = resource_type or self._get_resource_type()
-        return get_api_version(self.cli_ctx, resource_type)
+        version = get_api_version(self.cli_ctx, resource_type)
+        if isinstance(version, str):
+            return version
+        else:
+            version = getattr(version, operation_group, None)
+            if version:
+                return version
+            else:
+                from azure.cli.core.profiles._shared import APIVersionException
+                raise APIVersionException(operation_group, self.cli_ctx.cloud.profile)
```

Comment:
Technically the `else` isn't needed right?

Reply:
Well, otherwise it just returns None, which isn't terribly helpful.

Reply:
@derekbekoe meant you don't need the `else` but the `raise` is still needed. Since the control flow will exit this method on line 281 anyway the `else` is redundant.